### PR TITLE
rework update cashflow

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
Updates the submodule ID for the CashFlow submodule.

Also pulls in some updates from devel.

### CODE REVIEW

**Comment by PaulTalbot-INL** [comment]
Tests changed:
```
**** SUMMARY ****
*********************************
**** THE FOLLOWING TESTS CHANGED OUTCOMES ****
**** Test plugins/CashFlow.CashFlow_IRR | has changed outcome from <FAILED> to <OK> !!!!
**** Test plugins/CashFlow.CashFlow_NPV | has changed outcome from <FAILED> to <OK> !!!!
**** Test plugins/CashFlow.CashFlow_NPVsearch | has changed outcome from <FAILED> to <OK> !!!!
**** Test plugins/CashFlow.CashFlow_PI | has changed outcome from <FAILED> to <OK> !!!!
*****************
Number of success first file:
472
Number of success second file:
476
NUMBER OF SUCCESSES INCREASED: +4
*********************************
```

File: doc/user_manual/existing_interfaces.tex [hunk 1]
```diff
@@ -76,6 +76,30 @@ \subsection{Generic Interface}
 ...
 \end{lstlisting}
 
+
+%%%%
+
+If a code has hard-coded output file names that are not changeable,
```
Comment (wangcj05) [inline doc/user_manual/existing_interfaces.tex]: ```filenames```?

File: doc/user_manual/existing_interfaces.tex [hunk 2]
```diff
@@ -76,6 +76,30 @@ \subsection{Generic Interface}
...
+the GenericCode interface can be invoked using the \xmlNode{outputFile}
+node in which the output file name (CSV only) must be specified.
+For example, in the previous example, say instead of \texttt{-a gen.two} and \texttt{-o myOut}
+in the command line, the code always produce a CSV file named ``fixed@output.csv'';
```
Comment (wangcj05) [inline doc/user_manual/existing_interfaces.tex]: ```fixed@output.csv``` should be ```fixed_output.csv```?

File: framework/CodeInterfaces/Generic/GenericCodeInterface.py [hunk 3]
```diff
@@ -45,6 +45,22 @@ def __init__(self):
     self.execPrefix       = ''       # executioner command prefix (e.g., 'python ')
     self.execPostfix      = ''       # executioner command postfix (e.g. -zcvf)
     self.caseName         = None     # base label for outgoing files, should default to inputFileName
+    self.fixedOutFileName = None     # CSV output file name of the run code (in case it is hardcoded in the driven code)
```
Comment (wangcj05) [inline framework/CodeInterfaces/Generic/GenericCodeInterface.py]: maybe ```filename```?

File: framework/CodeInterfaces/Generic/GenericCodeInterface.py [hunk 4]
```diff
@@ -132,9 +148,10 @@ def getFileWithExtension(fileList,ext):
     #FIXME I think if you give multiple output flags this could result in overwriting
     self.caseName = inputFiles[index].getBase()
     outfile = 'out~'+self.caseName
-    if 'output' in clargs.keys():
+    if 'output' in clargs:
       todo+=' '+clargs['output']+' '+outfile
-    #text flags
+    if self.fixedOutFileName is not None:
+      outfile = self.fixedOutFileName
```
Comment (wangcj05) [inline framework/CodeInterfaces/Generic/GenericCodeInterface.py]: ```outFile```?

File: framework/Models/EnsembleModel.py [hunk 5]
```diff
@@ -374,7 +374,8 @@ def __selectInputSubset(self,modelName, kwargs):
     selectedkwargs['SampledVars'], selectedkwargs['SampledVarsPb'] = {}, {}
     for key in kwargs["SampledVars"].keys():
       if key in self.modelsDictionary[modelName]['Input']:
-        selectedkwargs['SampledVars'][key], selectedkwargs['SampledVarsPb'][key] =  kwargs["SampledVars"][key],  kwargs["SampledVarsPb"][key] if 'SampledVarsPb' in kwargs.keys() else 1.0
+        selectedkwargs['SampledVars'][key]   = kwargs["SampledVars"][key]
+        selectedkwargs['SampledVarsPb'][key] = kwargs["SampledVarsPb"][key] if 'SampledVarsPb' in kwargs.keys() and key in kwargs["SampledVarsPb"].keys() else 1.0
```
Comment (wangcj05) [inline framework/Models/EnsembleModel.py]: This already fixed by Diego, right?

Comment (wangcj05) [reply framework/Models/EnsembleModel.py]: I think the fix is only in devel.

**Comment by PaulTalbot-INL** [comment]
So, all of these changes came from those merged into "devel".  Do we want to modify those changes in this PR or make a PR to change them on devel?

**Comment by wangcj05** [comment]
The changes are from devel, basically we need to create a new PR for devel to fix these comments. I think we can merge this PR as it is. @alfoa What do you think?

**Comment by alfoa** [comment]
Good to me.

**Comment by wangcj05** [comment]
ok, good to go.

File: doc/user_manual/existing_interfaces.tex (hunk 1 above)
Comment (wangcj05) [reply doc/user_manual/existing_interfaces.tex]: fixed in PR#567

File: doc/user_manual/existing_interfaces.tex (hunk 2 above)
Comment (wangcj05) [reply doc/user_manual/existing_interfaces.tex]: fixed in PR#567

File: framework/CodeInterfaces/Generic/GenericCodeInterface.py (hunk 3 above)
Comment (wangcj05) [reply framework/CodeInterfaces/Generic/GenericCodeInterface.py]: fixed in PR#567

File: framework/CodeInterfaces/Generic/GenericCodeInterface.py (hunk 4 above)
Comment (wangcj05) [reply framework/CodeInterfaces/Generic/GenericCodeInterface.py]: fixed in PR#567

**Pull request merged**